src/whylogs/core/statistics/constraints.py

```diff
@@ -20,6 +27,77 @@
 
 logger = logging.getLogger(__name__)
 
+
+def _is_strftime_format(strftime_val, format):
```

Comment:
for these functions, your implementation is doing the expensive thing of actually parsing the object so let's make these utility functions a little more reusable (even if internal):
1) rename the prefix from "_is_" to "_try_parse_"
2) return the parsed object
3) add type hints for the input param and output (minor issue but helps readability)

Consider using optional return or tuple of (bool, concrete instance) e.g. Optional[datetime] or (datetime, bool)